sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LongLivedMessageSubscriber.java

```diff
@@ -0,0 +1,166 @@
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.reactivestreams.Subscription;
+import reactor.core.Disposable;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Mono;
+
+import java.util.Queue;
+import java.util.Timer;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class LongLivedMessageSubscriber extends BaseSubscriber<ServiceBusReceivedMessageContext> {
+
+    private final ClientLogger logger = new ClientLogger(SynchronousMessageSubscriber.class);
+    private Disposable timeoutOperation;
+    private final AtomicBoolean isDisposed = new AtomicBoolean(false);
+    private Queue<SynchronousReceiveWork> workQueue = new ConcurrentLinkedQueue<>();
+    private SynchronousReceiveWork currentWork = null;
+    private final AtomicInteger wip = new AtomicInteger();
+    private volatile Subscription subscription;
+    private long prefetch;
+
+
+    LongLivedMessageSubscriber(long prefetch) {
+        this.prefetch = prefetch;
```

Comment:
final

Reply:
I am planning to remove `prefetch` altogether. We are discussing this on teams.

Reply:
Why are you planning on removing prefetch? This would be your initial request in `hookOnSubscribe`